src/Compilers/Core/AnalyzerDriver/AnalyzerManager.AnalyzerAndOptions.cs

```diff
@@ -18,7 +18,13 @@ public AnalyzerAndOptions(DiagnosticAnalyzer analyzer, AnalyzerOptions analyzerO
                 Debug.Assert(analyzerOptions != null);
 
                 Analyzer = analyzer;
-                _analyzerOptions = analyzerOptions;
+                // _analyzerOptions is used only as a key for equality testing, and only the properties
```

Comment:
I think there are some VS only analyzer that depends on the implementation detail. what I mean is that it knows it is WorkspaceAnalyzerOption type and then try to cast to that type.